### PR TITLE
fix(134-2): Changed the alert and success banner to a more UX friendly design using toastify.

### PR DESCRIPTION
**Pull request by KevinTieu9**
Checklist:

- [ x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x ] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Changed the success banner to a more modern and more UX friendly design, the banner waits for a predefined timer before refreshing the page to show the user the new class

### CODE REVIEW

File: components/toastContainer.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+hideProgressBar={false}
+newestOnTop={false}
+closeOnClick
+rtl={false}
+pauseOnFocusLoss
+draggable
+pauseOnHover
+theme='light'
+/>
+{/* Same as */}
```
Comment (roberthuskins) [inline components/toastContainer.js:16]: get rid of comment

File: components/modal.js [hunk 2]
```diff
@@ -28,8 +32,24 @@ export default function Modal({ userId, certificationNames }) {
       method: 'POST',
       body: JSON.stringify(formData)
     });
-    router.reload();
-    alert('Successfully Created Class');
+
+  
+    //router.reload();
```
Comment (roberthuskins) [inline components/modal.js:37]: Why did we get rid of router.reload()?

File: components/modal.js [hunk 3]
```diff
@@ -28,8 +32,24 @@ export default function Modal({ userId, certificationNames }) {
...
+    //router.wait(10000);
+    setTimeout(function(){
+      window.location.reload();
+   }, 5000);
+    //alert('Successfully Created Class');
+
+    toast.success('Successfully Created Class!', {
```
Comment (roberthuskins) [inline components/modal.js:44]: Looks like a lot of these arguments in the function are being repeated by our `<ToastContainer>`. See my comment on the toastcontainer in the page.

File: components/modal.js [hunk 4]
```diff
@@ -28,8 +32,24 @@ export default function Modal({ userId, certificationNames }) {
...
-    router.reload();
-    alert('Successfully Created Class');
+
+  
+    //router.reload();
+    //router.wait(10000);
+    setTimeout(function(){
+      window.location.reload();
+   }, 5000);
+    //alert('Successfully Created Class');
```
Comment (roberthuskins) [inline components/modal.js:42]: get rid of these comments

File: components/toastContainer.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<ToastContainer />
```
Comment (roberthuskins) [inline components/toastContainer.js:17]: need to export this component like the others to be able to use it

File: pages/classes/index.js [hunk 6]
```diff
@@ -76,6 +79,19 @@ export default function Classes({
 }) {
   return (
     <>
+      <ToastContainer
```
Comment (roberthuskins) [inline pages/classes/index.js:82]: This <ToastContainer> is not imported from the components/toastContainer.js

The idea is you create a component under components that wraps the original ToastContainer to hide a lot of these options that we don't really want to configure every single time we want to add a ToastContainer. 
Let's call it "ClassroomToastContainer".

I don't think we need to configure options "position", closeonclick, hideProgressBar", "draggable" "rtl"... etc

I think we should be able only to pass in a few args, like color or whether the alert is a success and failure.

I don't know the exact answer to this problem, but essentially could you all go back and think about how to make this `<ToastContainer>` and `toast.success(` call to be as simple as possible for developers in the future? @GuillermoFloresV and I eventually want to refactor all of the alert(s to use your framework you are laying out here.

**Comment by GuillermoFloresV** [comment]
Hi, there @yeeao would it be possible for you to paste some pictures of the changes you've made so that we have more context to the changes? GitHub supports copy/pasting for pictures or even drag and drop, you can add them to the conversation or edit your original message to add them. Thanks!

**Comment by KevinTieu9** [comment]
[image: ClassSuccessfullyCreated.png]
[image: ClassSuccessfullyDeleted.png]
[image: somethingWrong.png]
I'll be getting rid of all the comments we made and clearing some of the
unneeded extras. I'll need to replace a portion of my code with
router.reload()

We were messing with toastContainer.js, I may end up deleting it in the
future if there is no use for it.

These are currently the three banners that will be changing based on
preferences, and will complete the other three by Saturday, if not, I would
like to continue this after the micro-internship ends.

Kevin Tieu

**Comment by yeeao** [comment]
> Hi, there @yeeao would it be possible for you to paste some pictures of the changes you've made so that we have more context to the changes? GitHub supports copy/pasting for pictures or even drag and drop, you can add them to the conversation or edit your original message to add them. Thanks!
will do, thanks for the advice!

**Pull request closed without merging**